underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -763,13 +763,15 @@ private boolean delete(String path, boolean recursive) throws IOException {
   private FileStatus[] listStatusInternal(String path) throws IOException {
     FileStatus[] files;
     FileSystem hdfs = getFs();
+    Path thePath = new Path(path);
     try {
-      files = hdfs.listStatus(new Path(path));
+      files = hdfs.listStatus(thePath);
     } catch (FileNotFoundException e) {
       return null;
     }
     // Check if path is a file
-    if (files != null && files.length == 1 && files[0].getPath().toString().equals(path)) {
+    if (files != null && files.length == 1 
+      && files[0].getPath().toUri().getPath().equals(thePath.toUri().getPath()) {
```

Comment:
indentation and a missing `)`
```suggestion
        && files[0].getPath().toUri().getPath().equals(thePath.toUri().getPath())） {
```